core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1426,7 +1426,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * supported for Hadoop-supported filesystems.
    */
   def addFile(path: String, recursive: Boolean): Unit = {
-    val uri = new URI(path)
+    val uri = new Path(path).toUri
```

Comment:
I do understand your concern @felixcheung. However,  IMHO, it'd be okay to not test Hadoop library within Spark. I will try to find some tests/documentation related with Windows path and then will share to make sure.

FWIW, this case was verified by one of comitters before for Windows path. So, it'd be okay.